backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/createFrom/po_from_so/impl/CreatePOLineFromSOLinesAggregator.java

```diff
@@ -176,14 +179,19 @@ protected void closeGroup(final I_C_OrderLine purchaseOrderLine)
 	{
 		final Set<OrderId> salesOrdersToBeClosed = new HashSet<>();
 		InterfaceWrapperHelper.save(purchaseOrderLine);
+
 		for (final I_C_OrderLine salesOrderLine : purchaseOrderLine2saleOrderLines.get(purchaseOrderLine))
 		{
-			salesOrderLine.setLink_OrderLine(purchaseOrderLine);
-			InterfaceWrapperHelper.save(salesOrderLine);
+			final I_C_PO_OrderLine_Alloc poAllocation = InterfaceWrapperHelper.newInstance(I_C_PO_OrderLine_Alloc.class);
+			poAllocation.setC_PO_OrderLine_ID(purchaseOrderLine.getC_OrderLine_ID());
+			poAllocation.setC_SO_OrderLine_ID(salesOrderLine.getC_OrderLine_ID());
+
+			orderDAO.savePOLineAllocation(poAllocation);
```

Comment:
Suggestion: extract this code into somthing like IOrderDAO.createNewAllocation. Maybe then you don't need IOrderDAO.savePOLineAllocation anymore?